File: src/entities/FuelItem.ts
export class FuelItem {
    x: number;
    y: number;
    radius: number = 30;
    collected: boolean = false;
    amount: number = 50; // Amount of fuel restored

    constructor(x: number, y: number) {
        this.x = x;
        this.y = y;
    }
}

File: unity_project/Assets/Scripts/Entities/FuelItem.cs
using UnityEngine;

namespace Entities {
    public class FuelItem : MonoBehaviour {
        public float Amount;
        public float Radius;
        public bool Collected;

        public void Init(float amount, float radius) {
            this.Amount = amount;
            this.Radius = radius;
            this.Collected = false;
        }

        public void Collect() {
            Collected = true;
            gameObject.SetActive(false);
        }
    }
}
